force-app/main/default/lwc/libsMomentjs/__tests__/libsMomentjs.test.js

```diff
@@ -0,0 +1,116 @@
+import { createElement } from 'lwc';
+import LibsMomentjs from 'c/libsMomentjs';
+import { loadScript } from 'lightning/platformResourceLoader';
+
+// The Lightning web component relies on data that is calculated
+// via the moment.js static resource. Instead of using the default
+// mock for loadScript we're creating a custom mock, that then loads
+// the JavaScript file from the static resource.
+//
+// This mock, if needed across multiple Lightning web components,
+// can also be added via moduleNameMapper in the jest.config.js
+// file.
+jest.mock(
+    'lightning/platformResourceLoader',
+    () => {
+        return {
+            loadScript() {
+                // eslint-disable-next-line no-unused-vars
+                return new Promise((resolve, reject) => {
+                    global.moment = require('../../../staticresources/moment');
+                    resolve();
+                });
+            }
+        };
+    },
+    { virtual: true }
+);
+
+// Sample error for loadScript error
+const LOAD_SCRIPT_ERROR = {
+    body: { message: 'An internal server error has occurred' },
+    ok: false,
+    status: 400,
+    statusText: 'Bad Request'
+};
+
+describe('c-libs-momentjs', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    // Helper function to wait until the microtask queue is empty. This is needed for promise
+    // timing when the platformResourceLoader promises.
+    function flushPromises() {
+        // eslint-disable-next-line no-undef
+        return new Promise(resolve => setImmediate(resolve));
+    }
+
+    it('loads the momentjs javascript static resource', () => {
+        const MOMENT_JS = 'moment';
+
+        // Create initial element
+        const element = createElement('c-libs-momentjs', {
+            is: LibsMomentjs
+        });
+        document.body.appendChild(element);
+
+        // Validation that the loadScript promise is called once.
+        // expect(loadScript).resolves.toBe(true);
+        // Validation that the chartjs static resource is passed as parameter.
+        return flushPromises().then(() => {
+            expect(loadScript.mock.calls[0][1]).toEqual(MOMENT_JS);
+        });
+    });
+
+    it('populates the disabled lightning-input fields with moment.js data based on user input', () => {
+        const INPUT_RAW = '2019-03-11T22:30:00.000Z';
+        const OUTPUT_EXPECTED = [70, 11];
```

Comment:
I like what this test is verifying, but I think it could be a little clearer in the test as to what exactly these values are. It took me a second and toggling over to the component a couple times to realize this is the 70th day and 11th week of 2019.

Reply:
Added comments to clarify test values.